apps/system/js/app_window.js

```diff
@@ -2069,6 +2071,10 @@
       return;
     }
 
+    if (this.isHomescreen) {
+      this.debug('die latency, die');
+      setTimeout(callback);
```

Comment:
shouldn't you have a `return;` just after? otherwise your callback can potentially be called multiple times.